standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3711,6 +3711,8 @@ public CompactionResponse compact(CompactionRequest rqst) throws MetaException {
             buf.append(thriftCompactionType2DbType(rqst.getType()));
             buf.append("',");
             buf.append(getEpochFn(dbProduct));
+            buf.append(", ?");
+            params.add(rqst.getPoolName());
```

Comment:
does it handle nulls properly?

Reply:
According to the tests in `org.apache.hadoop.hive.ql.txn.compactor.TestCompactionPools` yes.